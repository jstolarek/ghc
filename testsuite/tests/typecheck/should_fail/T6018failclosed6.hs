{-# LANGUAGE TypeFamilies, DataKinds, UndecidableInstances #-}

module T6018failclosed6 where

-- Non-injective type variable mentioned in the RHS
type family QClosed a b = r | r -> a where
    QClosed Int b = b
