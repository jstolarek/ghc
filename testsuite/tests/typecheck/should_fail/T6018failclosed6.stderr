
T6018failclosed6.hs:7:5:
    Type family equation violates injectivity annotation.
    Injective type variables ‘a’, ‘b’ do not appear in the RHS of type family equation
    Kind variable ‘k’ is not inferable from the RHS of type family equation
    forall (k :: BOX) (a :: k) (b :: k). Gc a b = Int
    In the equations for closed type family ‘Gc’
    In the type family declaration for ‘Gc’
